src/app/ngx-slimscroll/directives/slimscroll.directive.ts

```diff
@@ -69,7 +70,23 @@ export class SlimScrollDirective implements OnInit, OnDestroy {
     this.mutationThrottleTimeout = 50;
   }
 
-  ngOnInit() {
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.enabled) {
```

Comment:
I think `if (changes.enabled && this.enabled)` is shorter.

Reply:
It might be improper. With the current code, it will only do something when enabled changed. If applying your version, it will perform destruction every time changes happened. We have many inputs of the directive.

Reply:
okay.